linker/shared/src/test/scala/org/scalajs/linker/OptimizerTest.scala

```diff
@@ -425,6 +414,68 @@ class OptimizerTest {
       assertTrue(foundJSImport)
     }
   }
+
+  @Test
+  def testFoldLiteralClosureCaptures(): AsyncResult = await {
+    val classDefs = Seq(
+      mainTestClassDef({
+        consoleLog(Closure(true, List(paramDef("x", IntType)), Nil, None, {
+          BinaryOp(BinaryOp.Int_+, VarRef("x")(IntType), int(2))
+        }, List(int(3))))
+      })
+    )
+
+    for {
+      moduleSet <- linkToModuleSet(classDefs, MainTestModuleInitializers)
+    } yield {
+      traverseMainMethod(moduleSet) {
+        case c: Closure =>
+          assertTrue(s"captures remaining: $c", c.captureParams.isEmpty)
+          assertTrue(s"body is not a constant: $c", c.body.isInstanceOf[IntLiteral])
+        case _ =>
+      }
+    }
+  }
+
+  @Test
+  def testSupportImplicitClosureCaptures(): AsyncResult = await {
+    val calc = m("calc", Nil, I)
+
+    val classDefs = Seq(
+      classDef(
+          MainTestClassName,
+          kind = ClassKind.Class,
+          superClass = Some(ObjectClass),
+          memberDefs = List(
+              // @noinline static def calc(): Int = 1
+              MethodDef(EMF.withNamespace(PublicStatic), calc, NON, Nil,
+                  IntType, Some(int(1)))(EOH.withNoinline(true), None),
+              mainMethodDef(Block(
+                VarDef("x", NON, IntType, mutable = false,
+                    ApplyStatic(EAF, MainTestClassName, calc, Nil)(IntType)),
+                consoleLog(Closure(true, List(paramDef("x", IntType)), Nil, None, VarRef("x")(IntType), List(VarRef("x")(IntType))))
```

Comment:
Shouldn't you have chosen a different name for the capture param than its value, here?
```suggestion
                consoleLog(Closure(true, List(paramDef("y", IntType)), Nil, None, VarRef("y")(IntType), List(VarRef("x")(IntType))))
```

Reply:
True, that's even better.